Makefile

```diff
@@ -74,27 +75,22 @@ help: ## Print help for targets with comments.
 	@grep -E '^[a-zA-Z0-9._-]+:.*?## .*$$' Makefile | sort | \
 		awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-15s\033[0m %s\n", $$1, $$2}'
 
-# The .PHONY is needed to ensure that we recursively use the out/Makefile
-# to check for changes.
-.PHONY: $(NODE_EXE) $(NODE_G_EXE)
+$(NODE_EXE): out/Release/node
+	ln -fs $< $@
 
-# The -r/-L check stops it recreating the link if it is already in place,
-# otherwise $(NODE_EXE) being a .PHONY target means it is always re-run.
-# Without the check there is a race condition between the link being deleted
-# and recreated which can break the addons build when running test-ci
-# See comments on the build-addons target for some more info
-$(NODE_EXE): config.gypi out/Makefile
+$(NODE_G_EXE): out/Debug/node
+	ln -fs $< $@
+
+out/Release/node: out/Makefile $(ALL_DEPS)
 	$(MAKE) -C out BUILDTYPE=Release V=$(V)
-	if [ ! -r $@ -o ! -L $@ ]; then ln -fs out/Release/$(NODE_EXE) $@; fi
 
-$(NODE_G_EXE): config.gypi out/Makefile
+out/Debug/node: out/Makefile $(ALL_DEPS)
 	$(MAKE) -C out BUILDTYPE=Debug V=$(V)
-	if [ ! -r $@ -o ! -L $@ ]; then ln -fs out/Debug/$(NODE_EXE) $@; fi
 
-out/Makefile: common.gypi deps/uv/uv.gyp deps/http_parser/http_parser.gyp \
+out/Makefile .deps: deps/uv/uv.gyp deps/http_parser/http_parser.gyp \
               deps/zlib/zlib.gyp deps/v8/gypfiles/toolchain.gypi \
               deps/v8/gypfiles/features.gypi deps/v8/src/v8.gyp node.gyp \
-              config.gypi
+              common.gypi
```

Comment:
Is `config.gypi` intentionally dropped here?

Reply:
Yes, but it's no longer necessary.  I've pushed up a commit that restores it.

Reply:
I take back what I wrote - it's still necessary because of the `config.gypi: configure` rule below.  But since it's correct to depend on config.gypi here, I'll just remove the other rule.